### PR TITLE
feature(examples): adding net core x tracetest example

### PR DESCRIPTION
**Pull request by xoscar**
This PR includes a quick get-started example for Tracetest and .Net Core

## Changes

- Includes a new example folder

## Fixes

- #1975 

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/fedb4bdc9b4f421cabb5261c46e7bba4

### CODE REVIEW

File: examples/quick-start-net-core/collector.config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      http:
+
+processors:
+  batch:
+    timeout: 100ms
+
+  # Data sources: traces
+  probabilistic_sampler:
+    hash_seed: 22
+    sampling_percentage: 100
```
Comment (adnanrahic) [inline examples/quick-start-net-core/collector.config.yaml:14]: Is this section critical? I'd remove it.

File: examples/quick-start-net-core/src/Controllers/WeatherForecastController.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+namespace quick_start_net_core.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class WeatherForecastController : ControllerBase
+{
+  private static readonly string[] Summaries = new[]
+  {
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+    };
```
Comment (adnanrahic) [inline examples/quick-start-net-core/src/Controllers/WeatherForecastController.cs:14]: This `};` looks strange here.

**Review by adnanrahic — CHANGES_REQUESTED** [review]
Just some minor comments.

File: examples/quick-start-net-core/src/Controllers/WeatherForecastController.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    {
+      Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
+      TemperatureC = Random.Shared.Next(-20, 55),
+      Summary = Summaries[Random.Shared.Next(Summaries.Length)]
+    })
+    .ToArray();
+    activity?.SetTag("result", JsonSerializer.Serialize(result));
+
+    return result;
+  }
```
Comment (mathnogueira) [inline examples/quick-start-net-core/src/Controllers/WeatherForecastController.cs:38]: Is this generating any spans?

**Comment by mathnogueira** [comment]
Looks good to me. Just adjust the things adnan pointed out and it is good to merge

File: examples/quick-start-net-core/collector.config.yaml (hunk 1 above)
Comment (xoscar) [reply examples/quick-start-net-core/collector.config.yaml]: Done 😄

File: examples/quick-start-net-core/src/Controllers/WeatherForecastController.cs (hunk 2 above)
Comment (xoscar) [reply examples/quick-start-net-core/src/Controllers/WeatherForecastController.cs]: Done 😄

File: examples/quick-start-net-core/src/Program.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    .AddService(serviceName: serviceName, serviceVersion: serviceVersion);
+
+builder.Services.AddOpenTelemetryTracing(tracerProviderBuilder =>
+{
+  tracerProviderBuilder
+      .AddConsoleExporter()
+      .AddOtlpExporter(opt =>
+        {
+          opt.Protocol = OtlpExportProtocol.Grpc;
+          opt.Endpoint = new Uri("http://otel-collector:4317");
```
Comment (schoren) [inline examples/quick-start-net-core/src/Program.cs:24]: I think we need to add configs for max batch size and timeout so it works fast

**Review by schoren — CHANGES_REQUESTED** [review]

File: examples/quick-start-net-core/src/Program.cs (hunk 4 above)
Comment (adnanrahic) [reply examples/quick-start-net-core/src/Program.cs:24]: Could we also add a commented-out section that uses `.http` with otel on port `4318`? Just to make sure people know when to use which. Also writing comments pointing this out would be awesome.

**Review by danielbdias — APPROVED** [review]
It sounds fine to me!

File: examples/quick-start-net-core/src/Program.cs (hunk 4 above)
Comment (xoscar) [reply examples/quick-start-net-core/src/Program.cs:29]: Done and done 😄

**Review by adnanrahic — APPROVED** [review]
:shipit:

**Review by schoren — APPROVED** [review]

**Pull request merged**